app/models/sample.rb

```diff
@@ -187,4 +183,8 @@ def self.sort_columns
   def self.sort_column?(attr_name)
     sort_columns.include?(attr_name)
   end
+
+  def original_batch
+    Batch.find_by_id(original_batch_id)
+  end
```

Comment:
**suggestion**: this is a relation, so you could declare it as such in the model. then we can leverage activerecord to generate efficient SQL queries (e.g. `report.samples.preload(:original_batch)`):

```ruby
belongs_to :original_batch, class_name: "Batch", optional: true
```

Reply:
Solved in a previous comment :)